backend/coins/btc/transaction.go

```diff
@@ -120,24 +120,16 @@ func (account *Account) newTx(
 	return utxo, txProposal, nil
 }
 
-// SendTx creates, signs and sends tx which sends `amount` to the recipient.
-func (account *Account) SendTx(
-	recipientAddress string,
-	amount coin.SendAmount,
-	feeTargetCode accounts.FeeTargetCode,
-	selectedUTXOs map[wire.OutPoint]struct{},
-	_ []byte,
-) error {
-	account.log.Info("Signing and sending transaction")
-	utxo, txProposal, err := account.newTx(
-		recipientAddress,
-		amount,
-		feeTargetCode,
-		selectedUTXOs,
-	)
-	if err != nil {
-		return errp.WithMessage(err, "Failed to create transaction")
+// SendTx implements accounts.Interface.
+func (account *Account) SendTx() error {
+	unlock := account.activeTxProposalLock.RLock()
```

Comment:
> don't see much sense in trying to send the same tx/proposal from multiple goroutines.

even if that doesn't happen what's the point of using a more restrictive lock than needed?

when i see lock() i assume the var is written somewhere, which makes the code a bit less clear.

Reply:
true. but locks sometimes are used to serialize processes as well, not just a simple read/write var.

you could set activeTx to `nil` in the ned in this method. I guess it would make it writable then :)

Reply:
I did set it to nil before, but figured it made it more confusing, as a reader would wonder why it's not cleared in other ways (e.g. when leaving the send screen) :shrug:

Reply:
>  what's the point of using a more restrictive lock than needed?
> when i see lock() i assume the var is written somewhere

I guess using the less common RWLock might also contributed to that. The regular sync.Lock just [states](https://golang.org/pkg/sync/#Mutex) it's a simple "mutual exclusion lock". Nothing about reading or writing.